pkg/networking/cilium/reconciler/reconciler.go

```diff
@@ -148,6 +166,21 @@ func (r *Reconciler) upgrade(ctx context.Context, logger logr.Logger, client cli
 	return controller.Result{}, nil
 }
 
+func (r *Reconciler) upgradeConfig(ctx context.Context, logger logr.Logger, client client.Client, installation *cilium.Installation, spec *cluster.Spec) (controller.Result, error) {
```

Comment:
nit:
```suggestion
func (r *Reconciler) updateConfig(ctx context.Context, logger logr.Logger, client client.Client, installation *cilium.Installation, spec *cluster.Spec) (controller.Result, error) {
```

Reply:
again, a nit, so no big deal
In general, I think upgrade indicates more a "destructive" change (like changing the version that requires rolling out new node or pod) and update for changes that don't have such drastic side effects.

That said, one could argue that updating the config might also end up rolling new DS (or maybe restarting them) and we have no control over that.